### PR TITLE
[5.3] Filesystem::get with offset and limit

### PR DESCRIPTION
**Pull request by PedroFTW**
Minor modification based on this Issue: laravel/internals#111

### CODE REVIEW

File: src/Illuminate/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -28,14 +28,21 @@ public function exists($path)
      *
      * @param  string  $path
      * @param  bool  $lock
+     * @param  int $offset
+     * @param  int $limit
```
Comment (GrahamCampbell) [inline src/Illuminate/Filesystem/Filesystem.php]: bool|int

Comment (PedroFTW) [inline src/Illuminate/Filesystem/Filesystem.php]: Fixed it

**Comment by GrahamCampbell** [comment]
A more normal default would be `null` rather than `false`.

File: src/Illuminate/Filesystem/Filesystem.php [hunk 2]
```diff
@@ -28,14 +28,22 @@ public function exists($path)
      *
      * @param  string  $path
      * @param  bool  $lock
+     * @param  bool|int $offset
```
Comment (GrahamCampbell) [inline src/Illuminate/Filesystem/Filesystem.php]: int|null

File: src/Illuminate/Filesystem/Filesystem.php [hunk 3]
```diff
@@ -45,27 +53,35 @@ public function get($path, $lock = false)
...
-    public function sharedGet($path)
+    public function sharedGet($path, $offset, $limit)
     {
         $contents = '';
 
+        if (! $limit) {
+            $offest = 0;
+            $limit = 1048576;
+        }
+        $len = $offset + $limit;
```
Comment (GrahamCampbell) [inline src/Illuminate/Filesystem/Filesystem.php]: No need to define this, just use this straight bellow.

**Comment by taylorotwell** [comment]
No plans to add this.

File: src/Illuminate/Filesystem/Filesystem.php [hunk 4]
```diff
@@ -28,14 +28,22 @@ public function exists($path)
...
     {
         if ($this->isFile($path)) {
-            return $lock ? $this->sharedGet($path) : file_get_contents($path);
+            $params = [$path, null, null];
+            if ($limit) {
+                $params[] = $offset;
+                $params[] = $limit;
+            }
+
+            return $lock ? $this->sharedGet($path, $offset, $limit) : call_user_func_array('file_get_contents', $params);
```
Comment (BePsvPT) [inline src/Illuminate/Filesystem/Filesystem.php]: Maybe use file_get_contents($path, false, null, $offset, $limit) is enough?

File: src/Illuminate/Filesystem/Filesystem.php [hunk 5]
```diff
@@ -45,27 +53,35 @@ public function get($path, $lock = false)
...
+     * @param  bool|int $limit
      * @return string
      */
-    public function sharedGet($path)
+    public function sharedGet($path, $offset, $limit)
     {
         $contents = '';
 
+        if (! $limit) {
+            $offest = 0;
```
Comment (vlakoff) [inline src/Illuminate/Filesystem/Filesystem.php]: Just for the record: typo `offest`, should be `offset`.

**Pull request closed without merging**